src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -327,19 +354,47 @@ class RemoteMachineTrainingService implements TrainingService {
         return deferred.promise;
     }
 
-    public cleanUp(): Promise<void> {
+    public async cleanUp(): Promise<void> {
         this.log.info('Stopping remote machine training service...');
         this.stopping = true;
-
+        await this.cleanupConnections();
+        return Promise.resolve();
+    }
+    
+    /**
+     * stop gpu_metric_collector process in remote machine and remove unused scripts
+     */
+    private async cleanupConnections(): Promise<void> {
+        for (const [rmMeta, client] of this.machineSSHClientMap.entries()) {
+            let jobpidPath: string = path.join(this.getRemoteScriptsPath(), 'pid');
+            await SSHClientUtility.remoteExeCommand(`pkill -P \`cat ${jobpidPath}\``, client);
```

Comment:
First, it's good to do cleanup for remote machines. 
Two suggestions: 
1. You need to catch all the expections during cleanup. (Please notice cleanup SHOULD NOT throw expections) 
2. Set time out for clean up options.

Reply:
1. add try catch in cleanup()
2. set time out for 10s